src/mongoose_wpool.erl

```diff
@@ -9,35 +9,135 @@
+            CallTimeout = proplists:get_value(call_timeout, Opts, 5000),
+            ets:insert(?MODULE, #mongoose_wpool{name = {Type, Host, Tag},
+                                                strategy = Strategy,
+                                                call_timeout = CallTimeout}),
+            {ok, Pid};
+        Error ->
+            Error
+    end.
+
+stop(Type) ->
+    stop(Type, global).
+
+stop(Type, Host) ->
+    stop(Type, Host, default).
+
+stop(Type, Host, Tag) ->
+    ets:delete(?MODULE, {Type, Host, Tag}),
+    wpool:stop_sup_pool(make_pool_name(Type, Host, Tag)).
+
+get_worker(Type) ->
+    get_worker(Type, global).
+
+get_worker(Type, Host) ->
+    get_worker(Type, Host, default).
+
+get_worker(Type, Host, Tag) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [#mongoose_wpool{strategy = Strategy}] ->
+            Worker = wpool_pool:Strategy(make_pool_name(Type, Host, Tag)),
+            {ok, whereis(Worker)};
+        [] ->
+            {error, pool_not_started}
+    end.
+
+call(Type, Request) ->
+    call(Type, global, Request).
+
+call(Type, Host, Request) ->
+    call(Type, Host, default, Request).
+
+call(Type, Host, Tag, Request) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [#mongoose_wpool{strategy = Strategy, call_timeout = CallTimeout}] ->
+            wpool:call(make_pool_name(Type, Host, Tag), Request, Strategy, CallTimeout);
+        [] ->
+            {error, pool_not_started}
+    end.
+
+call(Type, Host, Tag, HashKey, Request) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [#mongoose_wpool{call_timeout = CallTimeout}] ->
+            wpool:call(make_pool_name(Type, Host, Tag), Request, {hash_worker, HashKey}, CallTimeout);
+        [] ->
+            {error, pool_not_started}
+    end.
+
+cast(Type, Request) ->
+    cast(Type, global, Request).
+
+cast(Type, Host, Request) ->
+    cast(Type, Host, default, Request).
+
+cast(Type, Host, Tag, Request) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [#mongoose_wpool{strategy = Strategy}] ->
+            wpool:cast(make_pool_name(Type, Host, Tag), Request, Strategy);
         [] ->
-            undefined
+            {error, pool_not_started}
     end.
 
-save_pool_settings(PoolName, PoolSettings) ->
-    ets:insert(pool_settings_tab(), PoolSettings#mongoose_worker_pool{name = PoolName}).
+cast(Type, Host, Tag, HashKey, Request) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [#mongoose_wpool{}] ->
+            wpool:cast(make_pool_name(Type, Host, Tag), Request, {hash_worker, HashKey});
+        [] ->
+            {error, pool_not_started}
+    end.
+
+get_pool_settings(Type, Host, Tag) ->
+    case ets:lookup(?MODULE, {Type, Host, Tag}) of
+        [PoolOpts] -> PoolOpts;
+        [] -> undefined
+    end.
 
-delete_pool_settings(PoolName) ->
-    ets:delete(pool_settings_tab(), PoolName).
+get_pools() ->
```

Comment:
I think it'd be more informative if the function returns more info about the pools than only the name.

Reply:
That's what it's supposed to be in the first iteration: https://redmine.erlang-solutions.com/issues/19818

Reply:
Ok, fair enough.